internal/provider/models/alerts/test/threshold_test.go

```diff
@@ -0,0 +1,548 @@
+						component_kind = "source"
+						component_id = mezmo_http_source.my_source.id
+						inputs = [mezmo_http_source.my_source.id]
+						name = "my threshold alert"
+						event_type = "log"
+						operation = "custom"
+						script = "function myFunc(a, e, m) { return a }"
+						event_timestamp = ".timestamp"
+						conditional = {
+							expressions = [
+								{
+									field = ".event_count"
+									operator = "greater"
+									value_number = 5000
+								}
+							],
+						}
+						subject = "Threshold Alert for Log event"
+						message = "You received a threshold alert for a Log event"
+						ingestion_key = "abc123"
+					}`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr(
+						"mezmo_threshold_alert.default_log", "id", regexp.MustCompile(`[\w-]{36}`),
+					),
+					StateHasExpectedValues("mezmo_threshold_alert.default_log", map[string]any{
+						"pipeline_id":                            "#mezmo_pipeline.test_parent.id",
+						"inputs.#":                               "1",
+						"inputs.0":                               "#mezmo_http_source.my_source.id",
+						"active":                                 "true",
+						"component_id":                           "#mezmo_http_source.my_source.id",
+						"component_kind":                         "source",
+						"conditional.%":                          "3",
+						"conditional.expressions.#":              "1",
+						"conditional.expressions.0.field":        ".event_count",
+						"conditional.expressions.0.operator":     "greater",
+						"conditional.expressions.0.value_number": "5000",
+						"conditional.logical_operation":          "AND",
+						"event_type":                             "log",
+						"event_timestamp":                        ".timestamp",
+						"ingestion_key":                          "abc123",
+						"name":                                   "my threshold alert",
+						"operation":                              "custom",
+						"script":                                 "function myFunc(a, e, m) { return a }",
+						"severity":                               "INFO",
+						"style":                                  "static",
+						"subject":                                "Threshold Alert for Log event",
+						"message":                                "You received a threshold alert for a Log event",
+						"window_duration_minutes":                "5",
+						"window_type":                            "tumbling",
+					}),
+				),
+			},
+		},
+	})
+}
+
+// WARNING: Mixing test that throw normal schema validation errors cannot be mixed with
+// tests where we bubble up errors with diag. This tends to cause teardown errors and
+// may be a bug.
+
+func TestThresholdAlert_root_required_errors(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		PreCheck:                 func() { TestPreCheck(t) },
+		ErrorCheck: func(err error) error {
```

Comment:
The use of `ErrorCheck` is a new approach, but we might use it in the future to test multiple regex failures at once.  The rub is that it's on the `TestCase` level, so there can only be 1 step.  It'd be ideal if it were on the `TestStep` level, but no.